packages/components/pie-divider/src/index.ts

```diff
@@ -34,10 +50,13 @@ export class PieDivider extends LitElement implements DividerProps {
         };
 
         return html`
-            <hr
-                data-test-id="pie-divider"
-                aria-hidden="true"
-                class="${classMap(classes)}" />`;
+            <div
+                id="${componentSelector}"
+                data-test-id="${componentSelector}"
+                class="${classMap(classes)}">
+                ${this.renderDividerLabel()}
+                <hr aria-hidden="true"/>
+            </div>`;
```

Comment:
Whilst this is pretty good, we end up unnecessarily rendering a `div` for cases where no label is provided. 

What if we returned either the standalone `hr` as before, or if there is a label, the div + 2 hrs and slot? 

It slightly complicates the render logic, but it reduces the overall markup of this component for use cases where no label is provided.

What do you think?

Reply:
Good idea! i will try to implement it

Reply:
It's a good idea but could complicate the styles a bit. Definitely worth giving it a go though, let us know how you get on!

Reply:
I also had some doubts precisely because of the styles, but as for me it looks good after changes

https://github.com/justeattakeaway/pie/blob/fb07b178582c62d20e71d076cf4fcea6cf6a713e/packages/components/pie-divider/src/divider.scss